File: common/src/main/scala/com/goexp/common/util/ContentLoader.scala
package com.goexp.common.util

import java.io.IOException
import java.nio.ByteBuffer
import java.nio.charset.Charset
import java.nio.file.{Files, Path}

object ContentLoader {
  def loadFromFile(path: Path, charset: Charset): String = {
    try {
      val bytes = Files.readAllBytes(path)
      val html = charset.decode(ByteBuffer.wrap(bytes)).toString
      return html
    } catch {
      case e: IOException =>
        e.printStackTrace()
    }
    null
  }
}

File: common/src/main/scala/com/goexp/common/util/ContentLoader.java
package com.goexp.common.util;

import java.io.IOException;
import java.nio.ByteBuffer;
import java.nio.charset.Charset;
import java.nio.file.Files;
import java.nio.file.Path;

public class ContentLoader {

    public static String loadFromFile(Path path, Charset charset) {
        try {
            var bytes = Files.readAllBytes(path);
            var html = charset.decode(ByteBuffer.wrap(bytes)).toString();

            return html;
        } catch (IOException e) {
            e.printStackTrace();
        }

        return null;
    }
}
